src/protocols/secure_channel/RendezvousParameters.h

```diff
@@ -59,11 +60,55 @@ class RendezvousParameters
 
     // Discriminators in RendezvousParameters are always long (12-bit)
     // discriminators.
-    bool HasDiscriminator() const { return mDiscriminator <= kMaxRendezvousDiscriminatorValue; }
-    uint16_t GetDiscriminator() const { return mDiscriminator; }
+    bool HasDiscriminator() const { return mHasDiscriminator; }
+
+    // Obtains the long version of the discriminator, or 0 if short.
+    // WARNING: This is lossy and a bad idea to use. The correct method to use
+    //          is GetSetupDiscriminator(). This method exists for public
+    //          API backwards compatibility.
+    uint16_t GetDiscriminator() const
+    {
+        if (mSetupDiscriminator.IsShortDiscriminator())
+        {
+            ChipLogError(Discovery,
+                         "Get RendezvousParameters::GetDiscriminator() called with SHORT discriminator (inconsistent). Using value "
+                         "0 to avoid crash! Call GetSetupDiscriminator() to avoid loss.");
+            return 0;
+        }
+
+        if (!mHasDiscriminator)
+        {
+            ChipLogError(Discovery,
+                         "Get RendezvousParameters::GetDiscriminator() called without discriminator in params (inconsistent). "
+                         "Using value 0 to avoid crash! Ensure discriminator is set!");
+            return 0;
+        }
+
+        return mSetupDiscriminator.GetLongValue();
+    }
+
+    SetupDiscriminator GetSetupDiscriminator() const
+    {
+        if (!mHasDiscriminator)
+        {
+            ChipLogError(Discovery,
+                         "Get RendezvousParameters::GetSetupDiscriminator() called without discriminator in params (inconsistent). "
+                         "Returning last!");
```

Comment:
I'm not really sure what this "Returning last" bit means...

Reply:
Fixed